exonum/src/helpers/fabric/key.rs

```diff
@@ -0,0 +1,56 @@
+use std::marker::PhantomData;
+use std::fmt;
+
+/// `Key` provides strongly typed access to data
+/// inside `Context`.
+pub struct Key<T> {
+    // These fields are public so that `config_key`
+    // macro works outside of this crate. It should be
+    // replaced with `const fn`, once it is stable.
+    #[doc(hidden)]
+    pub __name: &'static str,
+    #[doc(hidden)]
+    pub __phantom: PhantomData<T>,
+}
+
+// We need explicit `impl Copy` because derive won't work if `T: !Copy`.
```

Comment:
I wonder if we can use [`derivative`](https://crates.io/crates/derivative) for similar cases.

Reply:
Hm, interesting... I think this is a very rare case, so I would prefer to avoid pulling in a proc-macro just for this.